deepspeed/runtime/engine.py

```diff
@@ -199,6 +202,45 @@ def __init__(self,
         self.flatten = util_ops.flatten
         self.unflatten = util_ops.unflatten
 
+    def _in_aml(self):
+        # read and environment variable to detect if we are using an Azure ML environment
+        if 'AZUREML_EXPERIMENT_ID' in os.environ:
+            return True
+        else:
+            return False
+
+    def _set_environment_variables_for_nccl_backend(self,
+                                                    args,
+                                                    master_port=6105,
+                                                    verbose=True):
```

Comment:
It might be good to have a docstring here outlining what assumptions we make about the environment and the modifications that we make.

Reply:
We're essentially taking this code from AML (https://azure.github.io/azureml-web/docs/cheatsheet/distributed-training#environment-variables-from-openmpi) to do this. Ideally in the future AML will provide this logic for us via the AML SDK so we don't have to maintain this piece when it inevitably changes. In the docstring maybe let's link to this cheatsheet as a reference?

Reply:
Thanks for the feedback, Shaden and Jeff. Adding the link to the cheat sheet sounds good. I will add it as a docstring.

Reply:
I have added the docstring.

Reply:
Awesome!